### PR TITLE
Comments added

### PR DESCRIPTION
**Pull request by casperklein**
# Description

- Comments added
- ESMTP (implicit TLS) added

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Improvement (non-breaking change that does improve existing functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation (README.md or ENVIRONMENT.md or the documentation)
- [ ] If necessary I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by fbartels — APPROVED** [review]

**Review by wernerfred — CHANGES_REQUESTED** [review]
Not sure If this is really needed as Ports are imho absolute basics If someone wants to Host a Mailserver. (You have to Provide Them in the Client as well). Here it messes Up the Compose File although the Ports are Well documented all over the documentation.

Either way i can live with this Change but please avoid comments in the Same Line and Change Them to the Line above the Ports. This way the File can be parsed properly.

**Review by polarathene — DISMISSED** [review]
I don't mind it, it's an example config anyway and may be useful inline. What was the motivation for adding these into the config vs a comment linking to the docs which details the ports?

Do we not provide awareness well enough on this, is it just a quick refresher reference, or will a user unfamiliar with the ports wanting to setup mailserver make any sense of it? Whom does it benefit?

If the intention is to clarify the ports to someone who is wondering why the config has listed a bunch that someone might not know what they're for, add a 2nd comment linking to docs for the `ports`. The user is otherwise going to look these up anyway, or vaguely recalls them but unsure of which was for what protocol or implicit TLS vs STARTTLS.

**Comment by casperklein** [comment]
> is it just a quick refresher reference

Exactly. I found it handy, having the port description inline and not to have to look it up once in a while (when I am unsure).

> This way the File can be parsed properly.

What do you mean. This is valid yaml. I intentionally took this form, because I found it more clear in that case. (I had your variant before, but found this approach better).

**Comment by wernerfred** [comment]
> What do you mean. This is valid yaml. I intentionally took this form, because I found it more clear in that case. (I had your variant before, but found this approach better).

Sure it is valid. I just ran too often into issues that have their origin in inline comments. It is not easy to remove comments for example by deleting every line that starts with `#`. Also I had issues with inline comments when parsing yaml files with other scripts for example. 
That are surely "my" problems or the ones from ppl that do not just simply use the project but those problems can easily avoided by placing comments in separate lines.

Like I said, it is not a dealbreaker for me but if there is no real reason behind it I would welcome comments in separate lines. 😃

**Comment by polarathene** [comment]
> Like I said, it is not a dealbreaker for me but if there is no real reason behind it I would welcome comments in separate lines.

If your ok with it, I think your request for changes review status is blocking a merge?

**Review by aendeavor — APPROVED** [review]

**Comment by aendeavor** [comment]
I approve this as well:)

**Review by wernerfred — APPROVED** [review]

File: docker-compose.yml [hunk 1]
```diff
@@ -8,11 +8,14 @@ services:
...
     ports:
-      - "25:25"
-      - "143:143"
-      - "587:587"
-      - "993:993"
+      - "25:25"    # SMTP  (explicit TLS => STARTTLS)
+      - "143:143"  # IMAP4 (explicit TLS => STARTTLS)
+      - "465:465"  # ESMTP (implicit TLS)
+      - "587:587"  # ESMTP (explicit TLS => STARTTLS)
+      - "993:993"  # IMAP4 (implicit TLS)
```
Comment (williamdes) [inline docker-compose.yml:18]: Very nice change !
That said 3 ports are still missing
```yml
            - "110:110" # POP3
            - "995:995" # POP3
            - "4190:4190" # Sieve filters
```

Comment (wernerfred) [reply docker-compose.yml:18]: No not really, `docker-compose.yml` is a **not** feature complete example of the mailserver.
Keep in mind that by default POP3 and SIEVE are not enabled https://github.com/docker-mailserver/docker-mailserver/blob/a0f4a375124b3b3402d5639a7d422501c2011b97/mailserver.env#L107-L108 and https://github.com/docker-mailserver/docker-mailserver/blob/a0f4a375124b3b3402d5639a7d422501c2011b97/mailserver.env#L80-L81

Therefore imho it is correct to not list these ports here

Comment (williamdes) [reply docker-compose.yml:18]: Yes, but they could be added and commented ;)

**Pull request merged**